impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Radar.js

```diff
@@ -0,0 +1,284 @@
+        measures: model.measures.fieldIndexes
+      };
+
+      var visualData = this._buildVisualData(dataTable, visualMappings);
+      var radarIndicators = this._buildRadarIndicators(dataTable, visualMappings, visualData);
+      return {
+        indicator: radarIndicators,
+        records: this._buildRadarRecords(dataTable, visualData, radarIndicators)
+      };
+    },
+
+    _buildVisualData: function(dataTable, visualMappings) {
+      var categoriesList = this._createEntityList();
+      var seriesList = this._createEntityList();
+
+      var rowCount = dataTable.getNumberOfRows();
+
+      // Rows in visual space (where "fields"/columns and their values are the names and values of visual roles).
+      var visualRows = new Array(rowCount);
+
+      for(var rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+        var category = this._getRowVisualEntity(dataTable, rowIndex, visualMappings.category, categoriesList);
+        var series = this._getRowVisualEntity(dataTable, rowIndex, visualMappings.series, seriesList);
+
+        visualRows[rowIndex] = {
+          category: category,
+          series: series
+        };
+      }
+
+      return {
+        rows: visualRows,
+        categories: categoriesList,
+        series: seriesList
+      };
+    },
+
+    _createEntityList: function() {
+      var list = [];
+      list.set = Object.create(null);
+      return list;
+    },
+
+    _getRowVisualEntity: function(dataTable, rowIndex, colIndexes, entityList) {
+      var value = dataTable.getCompositeValue(rowIndex, colIndexes);
+      var key = O.getSameTypeKey(value);
+
+      var entity = O.getOwn(entityList.set, key);
+      if(entity == null) {
+        var label = dataTable.getCompositeFormattedValue(rowIndex, colIndexes);
+        entity = {
+          key: key,
+          value: value,
+          label: label,
+          rowIndexes: []
+        };
+        entityList.set[entity.key] = entity;
+        entityList.push(entity);
+      }
+
+      // Do not register the "virtual" null entity.
+      var isDataBound = colIndexes.length > 0;
+      if(isDataBound) {
+        entity.rowIndexes.push(rowIndex);
+      }
+
+      return entity;
+    },
+
+    _getEntityTableView: function(entity, dataTable) {
+      var view = new TableView(dataTable);
+      if(entity.rowIndexes.length > 0) {
+        // Not the null series case.
+        view.setSourceRows(entity.rowIndexes);
+      }
+
+      return view;
+    },
+
+    _buildRadarIndicators: function(dataTable, visualMappings, visualData) {
+      var indicators = [];
+
+      // One indicator/axis per category value x measure name
+
+      visualData.categories.forEach(function(category) {
+        var categoryTableView = this._getEntityTableView(category, dataTable);
+
+        // When  category is not mapped, do not include the category label.
+        // Example: "Cars ~ "
+        var categoryLabelPrefix = category.label ? (category.label + " ~ ") : "";
```

Comment:
Can be addressed on a follow-up PR. Should have a constant for this on the base class and be using it here as well as passing it to the various calls to `DataTable#getCompositeFormattedValue`.

Reply:
Addressed https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1760/files#diff-758fe27164ce4a5a8b841d0c48dc1052fb689ac73c22fff589a9ca9ba5151ec1R27